create new matrix page (#2203)
grommet/hpe-design-system: aries-site/src/pages/templates/matrix-navigation.mdx

import { DiagramPreview, Example } from '../../layouts';

> ## When a User would Use:

Author: ```suggestion
## When a User would Use
```

- Pages where the hierarchy isn't crucial to the user experience, and where browsing is encouraged. For ex: HPE.com
- User flows where a global nav bar is visible at all times.

### Matrix within Global Header Example
Global Header can provide easy access points to allow a user to jump from one section to another at all times. 

<Example caption="Diagram illustrating the matrix navigation within a global header." plain pad="small" previewWidth="large" height={{ min: 'small' }}>
<DiagramPreview src={{dark: "/templateImages/Matrix-Global-Header.svg", light: '/templateImages/Matrix-Global-Header-Invert.svg'}} alt="Diagram illustrating the matrix navigation within a global header." />
</Example>

### Matrix within App Example
Users can search for information within a set of cards to select or navigate to different access points in no particualr order.

<Example caption="Diagram illustrating the matrix navigation within an application." plain pad="small" previewWidth="large" height={{ min: 'small' }}>

> <DiagramPreview src={{dark: "/templateImages/matrix-app-example.svg", light: '/templateImages/matrix-app-example-invert.svg'}} alt="Diagram illustrating the matrix navigation within an application." />

Reviewer: I noticed a bug in "DiagramPreview" where the "light" src is being applied in dark mode. I'm fixing this in my PR for "side-to-side" nav. But just to prep for that change, the `-invert` svg should be for the "dark" mode src

Author: thanks Ill fix that

import { DiagramPreview, Example } from '../../layouts';

## When a User would Use

> - Pages where the hierarchy isn't crucial to the user experience, and where browsing is encouraged. For ex: HPE.com

Reviewer 2: ```suggestion
- Pages where the hierarchy isn't crucial to the user experience, and where browsing is encouraged. For example, [HPE.com](https://www.hpe.com/us/en/home.html) which presents a wide variety of content and allows a visitor to choose their own journey, relevant to their needs.
```

> - User flows where a global nav bar is visible at all times.

Reviewer 2: ```suggestion
- User flows where a global navigation bar is visible at all times. This allows a user quick access to any of the site's primary hubs.
```

> ### Matrix within a Music App Example

Reviewer 2: ```suggestion
### Example: Matrix Navigation within a Music App
```

> An example of a music app where the playlist is the hub and a song is the spoke. From the song, a user 

Reviewer 2: ```suggestion
The way a music application allows a user to choose their own journey is a great example where matrix navigation is employed. Songs, albums, artists, playlists all interrelate, but are relevant to different users at different times. Applying matrix navigation provides a user easy access to all related content at any given time. On a song screen, they may access the album the song appears on, the artist who performed it, and/or the playlists to which the song has been added. No matter the user's current need, matrix navigation provides easy access and choice.
```

can matrix navigate to the artist or album. 

<Example 

>   caption="Illustration showing how the user can matrix navigate to a song from an album within a music application." 

Reviewer 2: ```suggestion
  caption="A good example of matrix navigation is how within a music application a user has the flexibility to navigate from a playlist to a song, and then from that song to the artist, album, or even back to its playlist." 
```

  plain 
  pad="small"
  previewWidth="large"
  height={{ min: 'small' }}
>
  <DiagramPreview 
    src={{
      dark: "/templateImages/matrix-music-invert.svg",
      light: '/templateImages/matrix-music.svg'
    }} 
    alt="Diagram illustrating the matrix navigation in a music app." />
</Example>

### Matrix within App Example

> Users can search for information within a set of cards to select or navigate to different access points in no particualr order.

Reviewer 2: This sentence is confusing to me.

```suggestion
Users can search for information within a set of cards to select or navigate to different access points in no particular order.
```

<Example 

>   caption="Illustration showing how a user can navigate within an application." 

Reviewer: ```suggestion
  caption="Illustration showing how a user can matrix navigate within an application." 
```

>   caption="Illustration showing how a user can matrix navigate within an application." 

Reviewer 2: The caption should explain what the illustration/diagram is trying to explain. 

Another way to think about it is "if I didn't have the illustration, how would I describe the concept in words?" Or, "what is the key message/point a reader should takeaway from the diagram?"

To the latter, I'm not sure what this diagram is illustrating or what the key point is... Is it trying to show me that dashboard, applications, profile, and help are always accessible navigation points to me?

import { DiagramPreview, Example } from '../../layouts';

Matrix Navigation lets the user determine their own path by
jumping into particular areas of the screen, in no specific order. 
This allows them to directly access content of their choosing,
without having to go up and back down to get there.

## When a User would Use

> - Pages where the hierarchy isn't crucial to the user experience, and where browsing is encouraged. For example, [HPE.com](https://www.hpe.com/us/en/home.html) which presents a wide variety of content and allows a visitor to choose their own journey, relevant to their needs.

Reviewer: ```suggestion
- Pages where the hierarchy isn't crucial to the user experience and where browsing is encouraged. For example, [HPE.com](https://www.hpe.com/us/en/home.html) presents a wide variety of content and allows a visitor to choose their own journey, relevant to their needs.
```
This wasn't a complete sentence grammatically.

- User flows where a global navigation bar is visible at all times. This allows a user quick access to any of the site's primary hubs.

### Example: Matrix Navigation within a Music App

> The way a music application allows a user to choose their own journey is a great example where matrix navigation is employed. Songs, albums, artists, and playlists all interrelate but are relevant to different users at different times. Applying matrix navigation provides a user easy access to all related content at any given time. On a song screen, they may access the album the song appears on, the artist who performed it, and/or the playlists to which the song has been added. No matter the user's current need, matrix navigation provides easy access and choice.

Reviewer 2: ```suggestion
The way a music application allows a user to choose their own journey is a great example where matrix navigation is employed. Songs, albums, artists, and playlists all interrelate, but are relevant to different users at different times.

Applying matrix navigation provides a user easy access to all related content at any given time. On a song screen, they may access the album the song appears on, the artist who performed it, and/or the playlists to which the song has been added. No matter the user's current need, matrix navigation provides easy access and choice.
```

> The way a music application allows a user to choose their own journey is a great example where matrix navigation is employed. Songs, albums, artists, playlists all interrelate, but are relevant to different users at different times. Applying matrix navigation provides a user easy access to all related content at any given time. On a song screen, they may access the album the song appears on, the artist who performed it, and/or the playlists to which the song has been added. No matter the user's current need, matrix navigation provides easy access and choice.

Reviewer: ```suggestion
The way a music application allows a user to choose their own journey is a great example where matrix navigation is employed. Songs, albums, artists, and playlists all interrelate but are relevant to different users at different times. Applying matrix navigation provides a user easy access to all related content at any given time. On a song screen, they may access the album the song appears on, the artist who performed it, and/or the playlists to which the song has been added. No matter the user's current need, matrix navigation provides easy access and choice.
```

> can matrix navigate to the artist or album. 

Reviewer 2: ```suggestion
```

<Example 
  caption="A good example of matrix navigation is how within a music application a user has the flexibility to navigate from a playlist to a song, and then from that song to the artist, album, or even back to its playlist." 
  plain 
  pad="small"
  previewWidth="large"
  height={{ min: 'small' }}
>
  <DiagramPreview 
    src={{
      dark: "/templateImages/matrix-music-invert.svg",
      light: '/templateImages/matrix-music.svg'
    }} 
    alt="Diagram illustrating the matrix navigation in a music app." />
</Example>

### Matrix within App Example
An HPE application example of a user Matrix Navigating to different access points which allows them to determine their own path.

<Example 

>   caption="Illustration showing how a user has flexability in creating their own path in no particular order using Matrix Navigation." 

Reviewer 2: ```suggestion
  caption="Illustration showing how a user has flexibility in creating their own path in no particular order using Matrix Navigation." 
```
